src/fixed.jl

```diff
@@ -15,7 +15,13 @@ struct Fixed{T <: Signed, f} <: FixedPoint{T, f}
 
     # constructor for manipulating the representation;
     # selected by passing an extra dummy argument
-    Fixed{T, f}(i::Integer, _) where {T,f} = new{T, f}(i % T)
+    function Fixed{T, f}(i::Integer, _) where {T, f}
+        if f == bitwidth(T)
+            Base.depwarn("The `f`, which is the same as the number of rawtype bits, will be denied in the future.", :Fixed)
```

Comment:
```suggestion
            Base.depwarn("`Fixed` reserves one bit for the sign. Support for `f=$f` with raw type `T=$T` will be removed in a future release.", :Fixed)
```

To avoid a performance hit from the interpolation you might need the `@noinline` trick, i.e.,

```julia
@noinline dowarn(f, T) = Base.depwarn(...)
f == bitwidth(T) && dowarn(f, T)
```
(Probably not necessary if DCE runs before the allocation pass, I'm not sure of the order so you can check the LLVM code if you want to be sure.)